app/routes/reports.tsx

```diff
@@ -73,8 +97,89 @@ export default function Reports() {
 		};
 	}, [reports]);
 
+	const getSelectedReports = useMemo(() => {
```

Comment:
this function seems like it could be refactored. There are multiple if statements and filter function calls, but I'm pretty sure this can be streamlined.  Also, a new fuse is created everytime the function is called. Is this necessary? Filter results are separate from sort results. You can sort filtered results, for example. In this implementation they seem linked.